nservicebus/upgrades/5to6-moving-away-from-ibus.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Moving away from `IBus` in NServiceBus Version 6
+summary: Describes how to send messages without the IBus
+---
+
+Starting with Version 6, the `IBus` interface has been deprecated.
+
+The `IBus` interface was one of the key interfaces when using previous versions of NServiceBus. It provided access to many of the operations like sending messages, subscribing to events, and manipulating headers. The `IBus` interface was available through dependency injection in message handlers, any custom components that were registered in the container and through other mechanisms like the saga base class.
+
+Some scenarios involving `IBus` include: endpoint creation, sending messages during endpoint startup, sending messages from within message handlers, using the injected `IBus` in custom components also registered in the container. 
+
+## During Endpoint Creation
+
+In previous versions of NServiceBus to start a new instance of an endpoint, either the `Bus` static class or the `IStartableBus` interface was used. In Version 6 these two concepts have been replaced. More information can be found in the [hosting](/nservicebus/hosting/) section of the documentation. Instead of the `Bus` class, a new class called `Endpoint` helps to start the endpoint.
+
+`BusConfiguration` has been deprecated. Use `EndpointConfiguration` instead to configure and start the endpoint. 
+
+snippet: 5to6-endpoint-start-stop
+
+Starting the endpoint, provides access to `IEndpointInstance` which can be used to send messages during endpoint start up instead of using the `IBus` interface.
+
+## Sending messages outside message handlers
+
+A common use of `IBus` is to invoke bus operations outside of message handlers, such as sending a message from an ASP.NET request or from a client application. Instead of an `IBus` the `IEndpointInstance` offers all available bus operations outside the message processing pipeline. For example:
+
+snippet:5to6-endpoint-send-messages-outside-handlers
```

Comment:
@SimonCropp - How about I something this:

A common use of `IBus` is to invoke bus operations outside of message handlers, such as sending a message from an ASP.NET request or from a client application. Instead of an `IBus` the `IEndpointInstance` offers all available bus operations outside the message processing pipeline. The `IEndpointInstance` is available during the creation of the endpoint. This interface can then be injected into the container for scenarios where it is needed to send messages outside the handler.